src/middlewares/permissions.js

```diff
@@ -86,6 +88,50 @@ class PermissionMiddlewareCreator {
     };
   }
 
+  static _getRequestAttributes(request) {
+    const hasBodyAttributes = request.body && request.body.data && request.body.data.attributes;
+    return hasBodyAttributes
+      && new QueryDeserializer(request.body.data.attributes).perform();
+  }
+
+  // generate a middleware that will check that ids provided by the request exist
+  // whithin the registered scope
+  _ensureRecordIdsInScope(model) {
+    if (!model) throw new Error('missing model');
```

Comment:
This will be used by both `forest-expresss-sequelize` and `forest-expresss-mongoose`, thus what about `missing model in the smartAction middleware definition`, a quick check on either the documentation or the support should provide more details?